src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -501,7 +501,80 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
         reducingTransformListApply(rest.elems.length) {
           super.transform(localTyper.typedPos(tree.pos)(consed))
         }
+      //methods on Double
+      //new Predef.doubleToDouble(x).isNaN() -> java.lang.Double.isNaN(x)
+      //new Predef.doubleToDouble(x).isInfinite() -> java.lang.Double.isInfinity(x)
+      //methods on Number
+      //new Predef.doubleToDouble(x).byteValue() -> x.toByte()
+      //new Predef.doubleToDouble(x).shortValue() -> x.toShort()
+      //new Predef.doubleToDouble(x).intValue() -> x.toInt()
+      //new Predef.doubleToDouble(x).longValue() -> x.toLong()
+      //new Predef.doubleToDouble(x).floatValue() -> x.toFloat()
+      //new Predef.doubleToDouble(x).doubleValue() -> x.toDouble()
+      //
+      //methods on Float
+      //new Predef.float2Float(x).isNaN() -> java.lang.Double.isNaN(x)
+      //new Predef.float2Float(x).isInfinite() -> java.lang.Double.isInfinity(x)
+      //methods on Number
+      //new Predef.float2Float(x).byteValue() -> x.toByte()
+      //new Predef.float2Float(x).shortValue() -> x.toShort()
+      //new Predef.float2Float(x).intValue() -> x.toInt()
+      //new Predef.float2Float(x).longValue() -> x.toLong()
+      //new Predef.float2Float(x).floatValue() -> x.toFloat()
+      //new Predef.float2Float(x).doubleValue() -> x.toDouble()
```

Comment:
Should we handle conversions from the other primitive types? `1.longValue` has the same issue, it compiles to `int2Integer(1).longValue` (boxes), not `intWrapper(1).longValue` (doesn't box).

On the other hand, is it worth handling these conversions? Are people actually using `xValue` instead of `toX`?

Reply:
seems easy enough to do. will squash and add this on top to review